backend-tests/tests/test_decommission.py

```diff
@@ -0,0 +1,272 @@
+        invd = ApiClient(inventory.URL_DEV)
+        invm = ApiClient(inventory.URL_MGMT)
+
+        r = uc.call('POST',
+                    useradm.URL_LOGIN,
+                    auth=(user.name, user.pwd))
+        assert r.status_code == 200
+        utoken = r.text
+
+        aset = device.authsets[0]
+        change_authset_status(devauthm, aset.did, aset.id, 'accepted', utoken)
+
+        # request auth
+        body, sighdr = deviceauth.auth_req(aset.id_data,
+                                           aset.pubkey,
+                                           aset.privkey,
+                                           tenant_token)
+
+        r = devauthd.call('POST',
+                      deviceauth.URL_AUTH_REQS,
+                      body,
+                      headers=sighdr)
+        assert r.status_code == 200
+        dtoken = r.text
+
+        # wait for the device provisioning workflow to do its job
+        timeout = time.time() + 60
+        while time.time() < timeout:
+            r = invm.with_auth(utoken).call('GET',
+                                            inventory.URL_DEVICE,
+                                            path_params={'id': aset.did})
+            if r.status_code == 200:
+                break
+            else:
+                logger.debug("waiting for the device to be added to inventory...")
+                time.sleep(1)
+        else:
+            assert False, "device not added to the inventory"
+
+        # check if the device can access API by patching device inventory
+        payload = [
+                {
+                    "name": "mac",
+                    "value": "foo"
+                }
+        ]
+        r = invd.with_auth(dtoken).call('PATCH',
+                                    inventory.URL_DEVICE_ATTRIBUTES,
+                                    payload)
+        assert r.status_code == 200
+
+        # get all decommission_device WFs for reference
+        query = conductor.workflow_by_name_query(conductor.DECOMMISSIONING_DEVICE_WORKFLOW_NAME)
+        initial_wfs = conductor.get_workflows(params=query)
+
+        # decommission
+        r = devauthm.with_auth(utoken).call('DELETE',
+                          deviceauth_v2.URL_DEVICE.format(id=aset.did))
+
+        # check device is rejected
+        r = dc.with_auth(dtoken).call('GET',
+                                      deployments.URL_NEXT,
+                                      qs_params={'device_type': 'foo',
+                                                 'artifact_name': 'bar'})
+        assert r.status_code == 401
+
+        # check that the workflow completed successfully
+        timeout = time.time() + (60 * 5)
+        while time.time() < timeout:
+            wfs = conductor.get_workflows(params=query)
+            if wfs['totalHits'] == initial_wfs['totalHits'] + 1:
+                workflow = wfs['results'][0]
+                m = re.search('(?<=device_id=)\w+', workflow['input'])
```

Comment:
Please make it an string literal with `r'(?<=device_id=)\w+'`, to avoid a warning on execution:

```
  /tests/test_decommission.py:201: DeprecationWarning: invalid escape sequence \w
    m = re.search('(?<=device_id=)\w+', workflow['input'])
```

Reply:
I've replaced this one with something completely different (this was simply not working)